source/infrastructure/lib/aws-instance-scheduler-lambda.ts

```diff
@@ -0,0 +1,159 @@
+import {LambdaToDynamoDB, LambdaToDynamoDBProps} from "@aws-solutions-constructs/aws-lambda-dynamodb";
+import {Aws, RemovalPolicy} from "aws-cdk-lib";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as cdk from "aws-cdk-lib";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+import {Construct} from "constructs";
+import {Effect, PolicyStatement} from "aws-cdk-lib/aws-iam";
+export enum DeploymentType {
+  SOLUTIONS_S3, CDK_DEPLOY
+}
+export interface InstanceSchedulerLambdaProps {
+  deploymentType: DeploymentType
+
+  solutionVersion: string
+  solutionTradeMarkName: string
+  solutionsBucket?: cdk.aws_s3.IBucket
+
+  memorySize: number
+  schedulerRole: cdk.aws_iam.Role
+
+  /**
+   * Lambda Function environment variables
+   */
+  environment?: {
+    [key: string]: string;
+  }
+}
+export class CoreScheduler extends Construct {
+
+  public readonly lambdaFunction: lambda.Function;
+  public readonly configTable: dynamodb.Table
+  private readonly stateTable: dynamodb.Table
+  private readonly maintenanceWindowTable: dynamodb.Table
+
+  constructor(scope: Construct, id: string, props: InstanceSchedulerLambdaProps) {
+    super(scope, id);
+
+    const lambdaToDynamoDbConstruct = new LambdaToDynamoDB(this, 'instance-scheduler-lambda', this.lambdaToDynamoPropsFrom(props))
+
+    this.lambdaFunction = lambdaToDynamoDbConstruct.lambdaFunction;
+    this.stateTable = lambdaToDynamoDbConstruct.dynamoTable;
+
+    const cfnStateTable = this.stateTable.node.defaultChild as dynamodb.CfnTable
+    cfnStateTable.overrideLogicalId('StateTable')
```

Comment:
Another way to get the logical id how you want is to add it to `scope` instead of `this`.